packages/react-sdk/src/useObservable.ts

```diff
@@ -57,11 +57,17 @@ export function useObservable<T>(
     ) as ObservableConfig<T>
 
     const value = useSyncExternalStore(
-        (subscriber) => observable.subscribe(subscriber, { fireImediately: opts?.fireImmediately }),
-        () => observable.value,
+        (subscriber) =>
+            observable
+                ? observable.subscribe(subscriber, { fireImediately: opts?.fireImmediately })
+                : () => undefined,
+        () => observable?.value,
     )
 
     useEffect(() => {
+        if (!value) {
+            return
+        }
         if (isPersisted(value)) {
             if (value.status === 'loaded') {
```

Comment:
i was surprised you didn't have to add a !value check below in the use memo here:
```
const data = useMemo(() => {
        if (isPersisted(value)) {
            const { data, status } = value
```

Reply:
Hm. Good catch, the `as ObservableValue<T>` type assert in the return hides the possible error here. 

I'll revisit the type implementation later, we need a better way to distinguish between `PersistedObservable` <> `Observable` in the type level

Reply:
Tried using function overloading. https://github.com/river-build/river/commit/51ba8f3c7dc84fd49ca80016cd9d1bd44ea99709
but it's not worth the implementation cost: makes hard to reason about, error messages gets 10x worse...

Reply:
https://github.com/river-build/river/pull/775/commits/ee99997338f90a3fd25e683d1d300d0272d78355 should fix it.

Reply:
how did we get into a situation where the observable was undefined?

Reply:
```ts
Members.getMember(userId: string): Member | undefined
```
We're returning undefined because the `userId` may not be in the members Map
- the userId can be a malformed id
- the userId can represent a user that isn’t in the map

Reply:
Members.getMember should always return a Member
the member should have a `membership` param which should be 
![image](https://github.com/user-attachments/assets/abba1ffa-9a6f-473b-b9ba-a104ad51029a)

If you are still loading the space the `isInitalized` should flag be false
If you request a member thats not in the space `membership` should be `SO_UNSPECIFIED`
If you are observing a member while they leave the value should flip to `SO_LEAVE`

Reply:
https://github.com/river-build/river/pull/686/commits/4a2a7b82fe203ae17b8eb3fe9e9d4c9c5d2826b5 solved this (in the `sync-user-metadata` branch)